cpp/src/arrow/util/bit-util.cc

```diff
@@ -208,6 +229,70 @@ void UnalignedBitmapAnd(const uint8_t* left, int64_t left_offset, const uint8_t*
   writer.Finish();
 }
 
+void AlignedBitmapOr(const uint8_t* left, int64_t left_offset, const uint8_t* right,
```

Comment:
I also wonder whether we can factor out the common harness with `BitmapAnd`'s variations (perhaps by templatizing with one or two callables?).